app/screens/send-bitcoin-screen/scanning-qrcode-screen.tsx

```diff
@@ -82,13 +83,30 @@ export const ScanningQRCodeScreen: ScreenType = ({
     }
 
     try {
-      const { valid } = validPayment(data, tokenNetwork, client)
+      const { valid, lnurl } = validPayment(data, tokenNetwork, client)
       if (valid) {
-        if (index <= 1) {
-          navigation.replace("sendBitcoin", { payment: data })
+        if (lnurl) {
+          const lnurlParams = await getParams(lnurl)
+          switch (lnurlParams.tag) {
+            case 'withdrawRequest':
+              // TODO: lnurl-withdraw         
+              break
+            case 'payRequest':
+              if (index <= 1) {
+                navigation.replace("sendBitcoin", { payment: data })
+              } else {
+                navigation.navigate("sendBitcoin", { payment: data })
+              }
+              break
+          }
         } else {
-          navigation.navigate("sendBitcoin", { payment: data })
+          if (index <= 1) {
+            navigation.replace("sendBitcoin", { payment: data })
+          } else {
+            navigation.navigate("sendBitcoin", { payment: data })
+          }
```

Comment:
Are the changes here to skip doing anything when tag is `withdrawRequest`?

Not a fan of the duplicated navigation here.

Reply:
yes, when it's lnurl-withdraw we need to trigger `receive` flow and since it's async call we either wait for it to finish or we navigate for regular invoices/bitcoin/username.

Reply:
is there a better way? I've optimized this part so there's no unncessary multiple calls. it's much better now.